### PR TITLE
enh(multi-actions): render non-parallel tool calls properly

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

We are currently rendering all actions for a message as parallel tool calls. This is a "lie" and might confuse the model.
This change leverages the `step` field on actions so we can render the correct history to the model. Tools that were called in parallel are rendered as parallel calls, and tools that were called sequentially are rendered as sequential calls (i.e, the result of the previous call(s) is rendered before the next tool call(s))

I also reversed the first loops (instead of iterating from the end of the array and using `.unshift`, we're iterating from the start and using `.push`)

## Risk

Only affects MA

## Deploy Plan

N/A

### CODE REVIEW

**Review by PopDaph — APPROVED** [review]
🚀

File: front/lib/api/assistant/generation.ts [hunk 1]
```diff
@@ -274,39 +276,52 @@ export async function renderConversationForModelMultiActions({
...
       }
 
-      if (function_messages.length > 0) {
-        messages.unshift(...function_messages);
-      }
-      if (function_calls.length > 0) {
-        messages.unshift({
+      for (const step of steps) {
+        if (!step?.length) {
+          logger.error(
```
Comment (spolu) [inline front/lib/api/assistant/generation.ts:301]: Let's error here?

**Review by spolu — APPROVED** [review]
Looks very reasonable

File: front/lib/api/assistant/generation.ts (hunk 1 above)
Comment (fontanierh) [reply front/lib/api/assistant/generation.ts:301]: panicked

**Pull request merged**